magma/smart/smart_bits.py

```diff
@@ -497,7 +497,7 @@ def __init__(self, value=None):
         super().__init__(self)
         if value is None:
             value = type(self)._to_magma_()()
-        self._value = bits(value)
+        self._value = value
```

Comment:
@rsetaluri does this make sense? We can't do a `m.bits` conversion here because it will cause an error in the interface construction pipeline, see https://github.com/phanrahan/magma/runs/6312244536?check_suite_focus=true#step:5:1191

```
self = <magma.interface.IO object at 0x7f5fe9840ee0>
defn = _Test = DeclareCircuit("_Test", "I0", SmartBits[8, False], "O1", SmartBits[4, False], "O2", SmartBits[16, False])

    def bind(self, defn):
        if self._bound:
            raise Exception("Can not bind IO multiple times")
        for port in self._ports.values():
>           port.name.set_defn(defn)
E           AttributeError: 'AnonRef' object has no attribute 'set_defn'

magma/interface.py:454: AttributeError
```

Basically, calling `m.bits` on the port value will create an anon bits value, that's no longer a DefnRef.  Based on this change, the original code was doing a NOOP anyways since value was an instance of Bits, so removing the conversion should be okay?